File: py-opsml/src/scouter/queue.rs
use pyo3::prelude::*;
use scouter_client::*;

pub fn add_queue_module(m: &Bound<'_, PyModule>) -> PyResult<()> {
    m.add_class::<SpcRecord>()?;
    m.add_class::<PsiRecord>()?;
    m.add_class::<CustomMetricRecord>()?;
    m.add_class::<LLMRecord>()?;
    m.add_class::<ServerRecord>()?;
    m.add_class::<ServerRecords>()?;
    m.add_class::<RecordType>()?;

    m.add_class::<Feature>()?;
    m.add_class::<Features>()?;
    m.add_class::<Metric>()?;
    m.add_class::<Metrics>()?;
    m.add_class::<EntityType>()?;

    // queue
    m.add_class::<QueueBus>()?;
    m.add_class::<ScouterQueue>()?;

    Ok(())
}

File: py-opsml/python/opsml/scouter/queue.py
# type: ignore
# pylint: disable=no-name-in-module

from . import queue as _queue_impl

ScouterQueue = _queue_impl.ScouterQueue
Queue = _queue_impl.Queue
KafkaConfig = _queue_impl.KafkaConfig
RabbitMQConfig = _queue_impl.RabbitMQConfig
RedisConfig = _queue_impl.RedisConfig
SpcServerRecord = _queue_impl.SpcServerRecord
PsiServerRecord = _queue_impl.PsiServerRecord
CustomMetricServerRecord = _queue_impl.CustomMetricServerRecord
ServerRecord = _queue_impl.ServerRecord
ServerRecords = _queue_impl.ServerRecords
Feature = _queue_impl.Feature
Features = _queue_impl.Features
RecordType = _queue_impl.RecordType
Metric = _queue_impl.Metric
Metrics = _queue_impl.Metrics
EntityType = _queue_impl.EntityType
LLMRecord = _queue_impl.LLMRecord

__all__ = [
    "ScouterQueue",
    "Queue",
    "KafkaConfig",
    "RabbitMQConfig",
    "RedisConfig",
    "SpcServerRecord",
    "PsiServerRecord",
    "CustomMetricServerRecord",
    "ServerRecord",
    "ServerRecords",
    "Feature",
    "Features",
    "RecordType",
    "Metric",
    "Metrics",
    "LLMRecord",
    "EntityType",
]
